### ISSUE TITLE
Set-AtwsContact :: Cannot convert Parameter -id from int64[] to int64

### ISSUE THREAD

**Issue by MitchellVanHorn** [officecenter/Autotask#29]
When running Set-AtwsContact with the -id parameter, I'm receiving the below error. Passing by -InputObject still works flawlessly.

Error converting Int64 Array to Int64. Line 621 of Dynamic\Set-AtwsContact.ps1

```
PS C:\> Set-AtwsContact -id <someIDNumber> -Active 0
Exception setting "Id": "Cannot convert the "System.Int64[]" value of type "System.Int64[]" to type "System.Int64"."
At C:\Users\mrtadmin\Documents\WindowsPowershell\Cache\2946\Dynamic\Set-AtwsContact.ps1:621 char:21
+           $Object.$($Parameter.Key) = $Value
+                     ~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], SetValueInvocationException
    + FullyQualifiedErrorId : ExceptionWhenSetting
```

**Comment by klemmestad** [comment]
This is definitely a bug. I have checked the code and there is definitely a mistake there. It will be fixed in the next version.

I am glad you use -InputObject, though. Because the API is the way it is you will always get faster results using -InputObject if you are trying to change or create more than 1 object at a time. This is true for all functions that accept the parameter. PowerShell is usually faster when you use the pipeline, but working with a remote SOAP endpoint you will want to collect and send any changes in batches. If you use the pipeline Powershell will automatically separate any objects and make an API call pr object. When you use -InputObject my code can update or create multiple objects in a single call.

This is not really relevant in this case, because this happens when you use -Id as well :) Or would have, if it worked... :(

**Comment by MitchellVanHorn** [comment]
Hmm... Well, my desire was not so much to pipeline ID's (thus, basically causing the serial execution you mentioned) -- but instead to pass an array of id's.

For instance, in this case, I was using Get-AtwsContact to pull the contacts from a specific client, then get a table of the Contacts Object ID, first name, lastname, and email address - then manually set expired accounts to inactive by running one Set-AtwsContact command, with -id <FirstID>,<SecondID>,<ThirdID>,etc (versus one at a time through the web portal). It's would just be a faster way for me to do some manual cleanup of our clients contacts, since the deactivation of contacts has been probably the most neglected thing in our environment. 

Would passing an Array of IDs to -id still cause the one-at-a-time update calls? Or would this batch changes together the same way as the -InputObject parameter?

example:
```
PS C:\> Get-AtwsContact -accountID <SomeAccountID> -Active 1 | Select id,FirstName,LastName,EmailAddress

      id FirstName LastName   EMailAddress
      -- --------- --------   ------------
29693951 Dude      Person1    DPerson1@domain.com
29698211 Another   Person2    APerson2@domain.com
29703411 Other     Example    OExample@domain.com

PS C:\> Set-AtwsContact -id 29698211,29703411 -active 0
```

**Comment by klemmestad** [comment]
Passing an array of IDs causes a single Get (Get-AtwsContact -Id <array of IDS>) and a single Set as if you used -InputObject. The fact that it fails is a bug and needs to be fixed. In terms of API calls the cheapest method would be to add all contacts you wish to inactivate to an array and pass that array to Set-AtwsContact. Something like
```powershell
$Contacts = Get-AtwsContact -accountID <SomeAccountID> -Active 1
$ContactsToInactivate = @()
Foreach ($Contact in $Contacts) {
 If ($Contact <satisfies some criteria>) {
    $ContactsToInactivate += $Contact 
  }
}

Set-AtwsContact -InputObject $ContactsToInactivate -active 0
```
But like I said, your code should have worked. I have a fix in the works, and I hope to publish it this week.

**Comment by klemmestad** [comment]
Solved from version 1.6.2.12.
